src/libfetchers/github.cc

```diff
@@ -300,7 +300,7 @@ struct GitLabInputScheme : GitArchiveInputScheme
         if ("PAT" == token.substr(0, fldsplit))
             return std::make_pair("Private-token", token.substr(fldsplit+1));
         warn("Unrecognized GitLab token type %s",  token.substr(0, fldsplit));
```

Comment:
Should this warning be removed then? If these tokens are actually fine, then we shouldn't spam the user with useless warnings.

Reply:
While tokens other than PAT and OAuth2 are not necessarily bad, they are also not confirmed to be working out of the box. Some tokens are not allowed to call the necessary endpoints.

The warning might also be useful for detecting typos of users who wanted to write one of the known token types.

So it would be reasonable to keep the warning, while allowing users to use any token type if they know what they're doing.